task-list/app/controllers/tasks_controller.rb

```diff
@@ -1,28 +1,63 @@
 class TasksController < ApplicationController
+
   def index
     @tasks = Task.all
     @title= "Task List"
   end
+
   def show
     id = params[:id]
     @task = Task.find(id)
     @title = @task.name
   end
+
   def new
     @title = "Create A New Task"
     @task = Task.new
   end
+
   def create
-    @task = Task.create(task_params)
+    task = Task.create(task_params)
     redirect_to "/"
   end
+
   def destroy
     id = params[:id]
-    @task = Task.find(id)
-    @task.delete
+    task = Task.find(id)
+    task.delete
+    redirect_to "/"
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+    render :edit_form
+  end
+
+  def update
+    id = params[:id]
+    task = Task.find(id)
+    task.update_attributes(task_params)
+    redirect_to "/"
+  end
+
+  def complete
```

Comment:
Could the `complete` and `incomplete` methods be turned into a single route, which toggles between the two values based on whether the `Task` from the DB is already completed or not?